src/Backups/BackupCoordinationStageSync.cpp

```diff
@@ -102,8 +133,38 @@ BackupCoordinationStageSync::State BackupCoordinationStageSync::readCurrentState
         if (!zk_nodes_set.contains("current|" + host + "|" + stage_to_wait))
         {
             UnreadyHostState unready_host_state;
-            unready_host_state.started = zk_nodes_set.contains("started|" + host);
-            unready_host_state.alive = zk_nodes_set.contains("alive|" + host);
+            const String started_node_name = "started|" + host;
+            const String alive_node_name = "alive|" + host;
+            const String alive_node_path = zookeeper_path + "/" + alive_node_name;
+            unready_host_state.started = zk_nodes_set.contains(started_node_name);
+
+            /// Because we do retries everywhere we can't fully rely on ephemeral nodes anymore.
+            /// Though we recreate "alive" node when reconnecting it might be not enough and race condition is possible.
+            /// And everything we can do here - just retry.
+            /// In worst case when we won't manage to see the alive node for a long time we will just abort the backup.
```

Comment:
How long is that time? Because other host can be alive, just trying to reconnect to ZooKeeper.

Reply:
The time is something like `backup_restore_keeper_max_retries * mean(min_backoff, max_backoff)`. With default values it is something like 20-30 seconds I guess. Actually I don't know how the backoff is calculated, whether is it exponential or not, but we can just double the number of retries. It will be still unreliable, but a little bit better.

Reply:
With settings in PR, it should be about 76 seconds for backoff timeout + operations duration
https://pastila.nl/?23e29d85/3c90d0785722bfb974a9f840af368d86

Reply:
76 seconds must be enough but I think an extra settings for that time would be better